c++/src/ColumnReader.cc

```diff
@@ -1696,13 +1690,25 @@ namespace orc {
   /**
    * Create a reader for the given stripe.
    */
-  std::unique_ptr<ColumnReader> buildReader(const Type& type, StripeStreams& stripe) {
+  std::unique_ptr<ColumnReader> buildReader(const Type& type, StripeStreams& stripe,
+                                            bool useTightNumericVector) {
     switch (static_cast<int64_t>(type.getKind())) {
-      case DATE:
-      case INT:
+      case SHORT: {
+        if (useTightNumericVector) {
+          return std::unique_ptr<ColumnReader>(
+              new IntegerColumnReader<ShortVectorBatch>(type, stripe));
+        }
+      }
+      case INT: {
```

Comment:
This seems to cause `implicit-fallthrough` warning and compilation failure.

Reply:
Should we fix it?

Reply:
Yes, we need to fix or mitigate this for Apache ORC 1.9.0 because this causes Docker Test failures.

Reply:
I filed a JIRA and PR for `branch-1.9` first. Will make another PR for main branch..
- https://issues.apache.org/jira/browse/ORC-1453
- https://github.com/apache/orc/pull/1548 (for branch-1.9)